handler.go

```diff
@@ -342,3 +310,51 @@ func closeSendStream(ctx context.Context, stream *yarpctransport.ClientStream) e
 	}
 	return nil
 }
+
+// streamIOHandler uses provided stream message reader to provide stream
+// IO methods to read requests one by one and handle responses.
+type streamIOHandler struct {
+	out             output
+	serializer      encoding.Serializer
+	streamMsgReader encoding.StreamRequestReader
+}
+
+// NextRequest returns the next stream request body from the given stream
+// message reader.
+// Note: this method must not be called once EOF has been returned.
```

Comment:
What happens if you call this method once EOF has been returned? would it panic? Please add this information.
You can also update the comment and mention the method return EOF once there is not message to read from the provided streamMsgReader

nit: you can use the style `It should be noted that calling this method once EOF was returned will....`

Reply:
It actually depends on the underlying reader of streamMsgReader, if it is STDIN then trying to read again after EOF would wait for data or EOF from STDIN. To make this much simpler, we can always return EOF from `NextRequest` if streamMsgReader ever returned EOF earlier.